x-pack/solutions/chat/plugins/workchat-app/public/application/components/chat/conversation/chat_conversation_progression.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { EuiTimelineItem, EuiLoadingElastic, EuiAvatar } from '@elastic/eui';
+import type { ProgressionEvent } from '../../../../../common/chat_events';
+import { FancyLoadingText } from '../../utilities/fancy_loading_text';
+
+interface ChatConversationMessageProps {
+  progressionEvents: ProgressionEvent[];
+}
+
+export const ChatConversationProgression: React.FC<ChatConversationMessageProps> = ({
+  progressionEvents,
+}) => {
+  if (progressionEvents.length === 0) {
+    return undefined;
+  }
+
+  const getText = (event: ProgressionEvent): string => {
+    switch (event.data.step) {
+      case 'planning':
+        return 'Thinking about which integration to use';
```

Comment:
i18n?

Reply:
I'm missing a lot of i18n in the components (and I will fix that), but for that particular one, I will it them hardcoded for now, as ultimately the translated labels will be coming from the server in the progression events.